From 6c5decbe806259c7d3f34146b81b72c520860a30 Mon Sep 17 00:00:00 2001
From: Gerald Combs <gerald@wireshark.org>
Date: Fri, 25 Jan 2013 20:17:00 +0000
Subject: [PATCH 4/6] Copy over r47114 by hand.

  ------------------------------------------------------------------------
  r47114 | eapache | 2013-01-16 06:14:05 -0800 (Wed, 16 Jan 2013) | 3 lines
  Changed paths:
     M /trunk/epan/proto.c

  Null the finfo pointer after we free it. Reduces (but does not completely fix)
  the errors from bug https://bugs.wireshark.org/bugzilla/show_bug.cgi?id=8197
  ------------------------------------------------------------------------

svn path=/trunk-1.8/; revision=47281
---
 epan/proto.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/epan/proto.c b/epan/proto.c
index 1d1b98c..1925969 100644
--- a/epan/proto.c
+++ b/epan/proto.c
@@ -575,6 +575,7 @@ proto_tree_free_node(proto_node *node, gpointer data _U_)
 		/* This is a child node. Don't free the per-tree data, but
 		 * do free the field_info data. */
 		FREE_NODE_FIELD_INFO(finfo);
+		node->finfo = NULL;
 	}
 
 #if 0
-- 
1.7.10.4

